### PR TITLE
Show an adblock admonition in the dev console

### PR DESCRIPTION
**Pull request by davidfischer**
This PR shows an adblock admonition in the browser developer console. While I don't think this will convert many hearts and minds, I believe it is a good first step.

screen shot 2018-04-02 at 10 24 10 pm

### CODE REVIEW

**Comment by davidfischer** [comment]
To test this locally, change `api_host` in `user_builds/<project>/rtd-builds/latest/_static/readthedocs-data.js` to `https://readthedocs.org` and load those docs.

File: readthedocs/core/static-src/core/js/doc-embed/sponsorship.js [hunk 1]
```diff
@@ -164,8 +178,12 @@ function init() {
                 promo.display();
             }
         },
-        error: function () {
+        error: function (xhr, textStatus, errorThrown) {
             console.error('Error loading Read the Docs promo');
+
+            if (xhr && xhr.status === 404 && rtd.api_host === 'https://readthedocs.org') {
```
Comment (ericholscher) [inline readthedocs/core/static-src/core/js/doc-embed/sponsorship.js]: Feel like we probably need more of a check here. Is there an error thrown when it's blocked? In chrome I see "Blocked by the client" -- is there a specific error we should be looking for that is similar?

Comment (davidfischer) [reply readthedocs/core/static-src/core/js/doc-embed/sponsorship.js]: From jQuery's perspective, it is just a 404.

**Comment by davidfischer** [comment]
I don't think this will do anything with respect to getting people to whitelist us. However, it illustrates how to detect an ad blocker and simply logs to the console (which might alert us to any false positives) before we take any further action.

File: readthedocs/core/static-src/core/js/doc-embed/sponsorship.js [hunk 2]
```diff
@@ -106,9 +106,23 @@ Promo.prototype.post_promo_display = function () {
...
+    console.log('Read the Docs hosts documentation for tens of thousands of open source projects.');
+    console.log('We fund our development (we are open source) and operations through advertising.');
+
+    console.log('We promise to:');
+    console.log(' - never let advertisers run 3rd party JavaScript');
+    console.log(' - never sell user data to advertisers or other 3rd parties');
+    console.log(' - only show advertisements of interest to developers');
+    console.log('You can read more about our approach to advertising here: https://docs.readthedocs.io/en/latest/ethical-advertising.html');
+
+    console.log('%cPlease whitelist *.readthedocs.io on your adblocker!', 'font-size: 2em');
```
Comment (humitos) [inline readthedocs/core/static-src/core/js/doc-embed/sponsorship.js]: I think we need to say `*.readthedocs.org` maybe?

Look at this

captura de pantalla_2018-04-04_12-07-37

Comment (humitos) [reply readthedocs/core/static-src/core/js/doc-embed/sponsorship.js]: it comes from EasyList

Comment (davidfischer) [reply readthedocs/core/static-src/core/js/doc-embed/sponsorship.js]: That's a good catch. I need to review this. We may need both...

Comment (davidfischer) [reply readthedocs/core/static-src/core/js/doc-embed/sponsorship.js]: This is a larger problem than I thought.

Whitelisting `*.readthedocs.io` is all that is necessary (on AdblockPlus at least) to allow Read the Docs ads on all of our sites hosted on a subdomain. However, ads will continue to be blocked on any custom domains where we host ads. Whitelisting `readthedocs.org` does not change this. This makes simple whitelisting impossible across all ~10,000 domains where we host ads.

Comment (davidfischer) [reply readthedocs/core/static-src/core/js/doc-embed/sponsorship.js]: After reviewing their filter syntax, we need the following two filters:

* `@@||readthedocs.org/*` (exception for making an ad request)
* `@@readthedocs.io^$document` (exception for hiding elements)

**Comment by davidfischer** [comment]
I noted the specific commands needed to whitelist across our custom domains.

**Pull request merged**